src/dmd/mtype.d

```diff
@@ -601,77 +612,81 @@ extern (C++) abstract class Type : ASTNode
         TypeFunction t2 = t.isTypeFunction();
 
         if (!t1 || !t2)
-            goto Ldistinct;
+            return distinct();
 
         if (t1.parameterList.varargs != t2.parameterList.varargs)
-            goto Ldistinct;
+            return distinct();
 
         if (t1.parameterList.parameters && t2.parameterList.parameters)
         {
             if (t1.parameterList.length != t2.parameterList.length)
-                goto Ldistinct;
+                return distinct();
 
             foreach (i, fparam1; t1.parameterList)
             {
                 Parameter fparam2 = t2.parameterList[i];
-
-                if (!fparam1.type.equals(fparam2.type))
+                void cov() { notcovariant |= !fparam1.isCovariant(t1.isref, fparam2); }
```

Comment:
```suggestion
                void checkParametersCovariant() { notcovariant |= !fparam1.isCovariant(t1.isref, fparam2); }
```

No need to keep the function name cryptic.

Reply:
done